### PR TITLE
Patch Initial Sync For Non Finalized Blocks

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] If a block has been previously processed, we allow the processing to occur in the event our chain head is 
behind. Validation of responses is done in the block fetcher, so it should be fine to filter it this way.
 
**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/sync/initial-sync/round_robin.go [hunk 1]
```diff
@@ -304,7 +304,8 @@ func (s *Service) isProcessedBlock(ctx context.Context, blk block.SignedBeaconBl
 	if err != nil {
 		return false
 	}
-	if blk.Block().Slot() <= finalizedSlot || (s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)) {
+	blockExistsInDB := s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)
+	if blk.Block().Slot() <= finalizedSlot || blockExistsInDB && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
```
Comment (prestonvanloon) [inline beacon-chain/sync/initial-sync/round_robin.go:308]:
```suggestion
	if blk.Block().Slot() <= finalizedSlot || (s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)) && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
```

This is kind of a nasty long line, but `blk.Block().Slot() <= finalizedSlot ` is true for most of initial sync. Putting this boolean first in the statement could reduce the number of db lookups.

Comment (terencechain) [inline beacon-chain/sync/initial-sync/round_robin.go:308]: Also a good candidate for a one liner comment so readers can quickly understand what this does

Comment (terencechain) [inline beacon-chain/sync/initial-sync/round_robin.go:308]: Curious on whether we have unit test for this as well

Comment (nisdas) [reply beacon-chain/sync/initial-sync/round_robin.go]: we do have a unit test, which correctly failed when I had an off by 1 :) https://github.com/prysmaticlabs/prysm/pull/9452/commits/09c3beeb34caf3cd50ef5b47ae3c8a1eea3d89da

**Review by rauljordan — APPROVED** [review]

**Pull request merged**